app/assets/javascripts/discourse/app/instance-initializers/video-placeholder.js

```diff
@@ -0,0 +1,54 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { iconHTML } from "discourse-common/lib/icon-library";
+
+export default {
+  initialize() {
+    withPluginApi("0.8.7", (api) => {
+      function _handleEvent(event) {
+        const svgIcon = event.target;
+        const overlay = svgIcon.parentElement;
+        const wrapper = overlay.parentElement;
+        const parentDiv = wrapper.parentElement;
+
+        wrapper.style.display = "none";
+
+        const videoHTML = `
+        <video width="100%" height="100%" preload="metadata" controls>
+          <source src="${parentDiv.dataset.videoSrc}" ${parentDiv.dataset.origSrc}>
+          <a href="${parentDiv.dataset.videoSrc}">${parentDiv.dataset.videoSrc}</a>
+        </video>`;
+        parentDiv.insertAdjacentHTML("beforeend", videoHTML);
+        parentDiv.classList.add("video-container");
+
+        const video = parentDiv.querySelector("video");
+        video.play();
+      }
+
+      function _attachCommands(post, helper) {
```

Comment:
`_attachCommands` is quite vague, I know we use it in a few other decorators, but I'd vote for a rename, something like `applyVideoPlaceholder`, maybe. 

Also, we have some video decorators in `post-decorations.js`,I wonder if this refactor breaks them or makes them irrelevant now (especially the iOS Safari one).